src/workergroup.ts

```diff
@@ -70,6 +70,9 @@ export class WorkerGroup {
             this.startedCallback();
           }
         } else if (data.type === "done") {
+          const puzzleIndex = this.puzzleSolverInputs.indexOf(data.puzzleSolverInput);
```

Comment:
Does this actually work? I would expect that `buf1 == buf2` to be false even though the data in it is the same (as it compares the ref, not the data within), after `buf2` was sent over a webworker message (unless it sends by reference, but I thought it would serialize the messages).

Reply:
I could of course be totally wrong!

Reply:
Yeah I already noticed that it doesn't work 😢

Reply:
Not too sure what the best identifier is to send with the data then. Maybe just add a counter to the `WorkerGroup` that is incremented when work on a new puzzle has started?

Reply:
Yeah that sounds like the simplest approach: the worker group has a counter that starts at 0, and whenever a new puzzle is started we increment it, in the callback we check whether it matches :).

The alternative is to loop over the data and compare, but I guess that's more code (every 0.1kb does make a difference of course).

Reply:
Ok, this does the trick 👍